new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodTemplates.java

```diff
@@ -0,0 +1,495 @@
+
+  private static String clusterName = "cluster-1";
+  private static String miiImage = null;
+  private static String wdtImage = null;
+  private static LoggingFacade logger = null;
+
+  /**
+   * Install operator.
+   *
+   * @param namespaces list of namespaces created by the IntegrationTestWatcher by the
+   *                   JUnit engine parameter resolution mechanism
+   */
+  @BeforeAll
+
+  public static void initAll(@Namespaces(3) List<String> namespaces) {
+
+    logger = getLogger();
+    // create standard, reusable retry/backoff policy
+    withStandardRetryPolicy = with().pollDelay(2, SECONDS)
+        .and().with().pollInterval(10, SECONDS)
+        .atMost(5, MINUTES).await();
+
+    logger.info("Get a unique namespace for operator");
+    assertNotNull(namespaces.get(0), "Namespace list is null");
+    final String opNamespace = namespaces.get(0);
+
+    logger.info("Get a unique namespace for WebLogic domain1");
+    assertNotNull(namespaces.get(1), "Namespace list is null");
+    domain1Namespace = namespaces.get(1);
+
+    logger.info("Get a unique namespace for WebLogic domain2");
+    assertNotNull(namespaces.get(2), "Namespace list is null");
+    domain2Namespace = namespaces.get(2);
+
+    logger.info("install and verify operator");
+    installAndVerifyOperator(opNamespace, domain1Namespace,domain2Namespace);
+
+    logger.info("create and verify WebLogic domain image using model in image with model files");
+    miiImage = MII_BASIC_IMAGE_NAME + ":" + MII_BASIC_IMAGE_TAG;
+  }
+
+  /**
+   * Test pod templates using all the variables $(SERVER_NAME), $(DOMAIN_NAME), $(DOMAIN_UID),
+   * $(DOMAIN_HOME), $(LOG_HOME) and $(CLUSTER_NAME) in serverPod for Domain In Image. Make sure the domain comes up
+   * successfully.
+   *
+   * @throws Exception when the domain crd creation fails or when updating the serverPod with
+   *                   variables
+   */
+  @Test
+  @DisplayName("Test pod templates using all the variables for domain in image.")
+  public void testPodTemplateUsingVariablesDomainInImage() throws Exception {
+    wdtImage = createAndVerifyDomainInImage();
+    try {
+      logger.info("Create wdt domain and verify that it's running");
+      createAndVerifyDomain(wdtImage, domain2Uid, domain2Namespace, "Image", replicaCount);
+      String managedServerPodName = domain2Uid + "-" + MANAGED_SERVER_NAME_BASE + "1";
+      V1Pod managedServerPod = Kubernetes.getPod(domain2Namespace, null, managedServerPodName);
+      assertNotNull(managedServerPod,"The admin pod does not exist in namespace " + domain2Namespace);
+      String serverName = managedServerPod
+          .getMetadata().getLabels()
+          .get("servername");
+      assertNotNull(serverName, "Can't find label servername");
+      assertTrue(serverName.equalsIgnoreCase("managed-server1"),
+          "Can't find or match label servername");
+      String domainName = managedServerPod
```

Comment:
throughout the test add some logs and comments as to what you are verifying.

Reply:
added